components/movementsensor/dualgps/dualgps.go

```diff
@@ -204,11 +204,24 @@ func (dg *dualGPS) Readings(ctx context.Context, extra map[string]interface{}) (
 	return movementsensor.DefaultAPIReadings(ctx, dg, extra)
 }
 
-// Unimplemented functions.
 func (dg *dualGPS) Position(ctx context.Context, extra map[string]interface{}) (*geo.Point, float64, error) {
-	return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(), movementsensor.ErrMethodUnimplementedPosition
+	dg.mu.Lock()
+	defer dg.mu.Unlock()
+	geoPoint1, alt1, err := dg.gps1.Position(ctx, nil)
+	if err != nil {
+		return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(), err
+	}
+	geoPoint2, alt2, err := dg.gps2.Position(ctx, nil)
+	if err != nil {
+		return geo.NewPoint(math.NaN(), math.NaN()), math.NaN(), err
```

Comment:
just a thought but if one of the gps returns an error but the other one is successful should we return the points of the one that works instead of giving the user NaN?

Reply:
works for mobile robots, but disastrous for something like a boat/drone. 

However, I like it, will do!